src/foremast/securitygroup/create_securitygroup.py

```diff
@@ -162,10 +199,13 @@ def add_cidr_rules(self, rules):
                 if 'InvalidPermission.Duplicate' in str(error):
                     self.log.debug('Duplicate rule exist, that is OK.')
                 else:
-                    msg = 'Unable to add cidr rules to {}'.format(rules['app'])
+                    msg = 'Unable to add cidr rules to {}'.format(rule.get('app'))
                     self.log.error(msg)
                     raise SpinnakerSecurityGroupError(msg)
 
+        # Tag security group
+        self.add_tags(group_id)
```

Comment:
I thought so too, I just placed there because group_id was already called. I moved the group_id call into function as well then placed in `create_security_group` which is correct place yes.